lib/wickedbase.pm

```diff
@@ -117,6 +119,104 @@ sub assert_wicked_state {
     $self->ping_with_timeout(ip => $args{ping_ip}) if $args{ping_ip};
 }
 
+sub valgrind_get_services {
+    my $self = shift;
+    my $val = get_var('WICKED_VALGRIND');
+
+    return if (!defined($val) || !$val);
+    my @all = qw(wickedd-auto4
+      wickedd-dhcp6
+      wickedd
+      wickedd-dhcp4
+      wickedd-nanny
+      wicked);
+    my @tmp;
+    my @enable;
+    if (index($val, ",")) {
+        @tmp = split(/,/, $val);
+    } else {
+        @tmp = $val;
+    }
+    foreach my $v (@tmp) {
+        $v =~ s/\.service$//;
+        if (grep { /^$v$/ } @all) {
+            push @enable, $v;
+        } elsif ($v =~ /^1|all$/i) {
```

Comment:
what "/^1|" part means ?

Reply:
if the value is `1` or `all`, then valgind get's enabled for every wicked related binary.